jmix-flowui/flowui-kit/src/main/java/io/jmix/flowui/kit/component/grid/GridMenuItemActionSupport.java

```diff
@@ -42,32 +48,72 @@ public Action getAction() {
     }
 
     public void setAction(@Nullable Action action) {
-        if (Objects.equals(this.action, action)) {
+        setAction(action, true);
+    }
+
+    public void setAction(@Nullable Action action, boolean overrideComponentProperties) {
+        if (Objects.equals(this.action, action) && this.overrideComponentProperties == overrideComponentProperties) {
             return;
         }
 
         removeRegistrations();
 
         this.action = action;
+        this.overrideComponentProperties = overrideComponentProperties;
 
         if (action != null) {
-            menuItem.setText(action.getText());
-            menuItem.setEnabled(action.isEnabled());
-            menuItem.setVisible(action.isVisible());
-
-            registration = menuItem.addMenuItemClickListener(event ->
+            GridMenuItem<?> contextMenuItem = menuItem.getMenuItem();
+            if (overrideComponentProperties) {
+                menuItem.setText(action.getText());
+                contextMenuItem.setEnabled(action.isEnabled());
+                contextMenuItem.setVisible(action.isVisible());
+                menuItem.setTooltipText(action.getDescription());
+                if (isShowActionIconEnabled()) {
+                    menuItem.setPrefixComponent(createIconComponent(action.getIcon()));
+                }
+                if (isShowActionShortcutEnabled()) {
+                    menuItem.setSuffixComponent(createShortcutComponent(action.getShortcutCombination()));
+                }
+                actionPropertyChangeRegistration = addPropertyChangeListener();
+            }
+            registration = contextMenuItem.addMenuItemClickListener(event ->
                     action.actionPerform(event.getSource()));
-            actionPropertyChangeRegistration = addPropertyChangeListener();
         }
 
         updateVisible();
     }
 
+    protected boolean isShowActionIconEnabled() {
+        return false;
+    }
+
+    @Nullable
+    protected Component createIconComponent(@Nullable Icon actionIcon) {
```

Comment:
Change `Icon` to `Component`, `SvgIcon` and `FontIcon` don't extend Icon, so let's make it more generic.